### PR TITLE
China plugin page update

### PR DESCRIPTION
**Pull request by langsmith**
This pr updates https://www.mapbox.com/android-docs/plugins/overview/china to bring the 📜 a bit more up-to-date, separate topics, fix typos, etc.

cc @springmeyer @jmkiley @lloydsheng @chriswu42 

screencapture-localhost-8080-android-docs-plugins-overview-china-2018-08-02-12_01_15

### CODE REVIEW

File: src/pages/plugins/overview/china.md [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 
-## Install the China Plugin
-You'll need to add the appropriate dependencies inside of your `build.gradle` file to start developing an application using the China Plugin. Both dependencies include the Maps SDK for Android. Version bumps of this plugin tend to follow after a new release of the Maps SDK for Android. All dependencies given below can be found on Bintray.
+## Install the China plugin
+
+You'll need to add the appropriate dependencies inside of your `build.gradle` file to start developing an application using the China Plugin. Both dependencies include the Maps SDK for Android. Version bumps of this plugin tend to follow after a new _stable_ release of the Maps SDK for Android that happens approximately every 4 weeks. All dependencies given below can be found on Bintray.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: follow ~~after~~ (redundant)

Comment (colleenmcginnis) [reply src/pages/plugins/overview/china.md]: Maybe rephrase altogether to: 

>Version bumps of this plugin tend to follow new _stable_ releases of the Maps SDK for Android, which happen approximately every 4 weeks.

File: src/pages/plugins/overview/china.md [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
+There are two dependency options, or flavors, available. The choice between the twom ultimately comes down to the location of the app's end user. The end user is either inside or outside of China. Using the `Global` flavor within China will result in slower speeds for fetching map tiles and inaccurate coordinates. Using the `China` flavor corrects for these issues in two major ways:
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: > twom

to 

>two

😝

File: src/pages/plugins/overview/china.md [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
+
+- Endpoints that the Mapbox APIs get their data from. The China flavor makes all API requests and receives map tiles from within China, making the app up to 10x faster.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: Let's make the first sentence a complete sentence: 

>It changes the endpoints that the Mapbox APIs get their data from.

File: src/pages/plugins/overview/china.md [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
+- The `China flavor includes the [shifting module](#shifting-annotations), whicallowing you to shift all of your map annotations to be accurately represented on top of the map. These flavors are interchangeable. Switching between the two requires no changes in code other than changing the dependency name in the your app-level 'build.gradle' file. This allows you the flexibility to have a single app for inside China, or an app with two equal flavors - one for within China and one for everywhere else.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: Forgot a closing ` mark after China.

Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: > whicallowing

to 

> which allow

Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: >dependency name in ~~the~~ your app-level

File: src/pages/plugins/overview/china.md [hunk 5]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
+The China Plugin for Android has wrapper classes to be used in place of their "typical" counterparts. For example, in the regular Maps SDK for Android, you'd use the `MapView` inside your activities' layouts. When using this plugin, a lint error will appear telling you to use the `ChinaMapView` instead, which wraps the `MapView` object. Under the hood, the `ChinaMapView` class does nothing more but set default values for optimal performance inside China. This includes requesting map tiles from our Chinese servers. In the provided chart below, you'll find a list of the objects which you'd typically use in the Maps SDK for Android and their counterparts found inside of this plugin.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: >In the ~~provided~~ chart below

File: src/pages/plugins/overview/china.md [hunk 6]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
 
 ## Using the correct objects
-The China Plugin for Android has wrapper classes to be used in place of their "typical" counterparts. For example, in the regular Maps SDK for Android, you'd use the `MapView` inside your activities' layouts. When using this plugin, a lint error will appear telling you to use the `ChinaMapView` instead, which wraps the `MapView` object. Under the hood, the `ChinaMapView` class does nothing more but set default values for optimal performance inside China. This includes hitting our inner China servers to serve up map tiles. In the provided chart below, you'll find a list of the objects which you'd typically use in the Map SDK for Android and their counterparts found inside of this plugin.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: >a list of the objects which you'd typically

which --> that

File: src/pages/plugins/overview/china.md [hunk 7]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
+As mentioned above, Android Lint will attempt to warn you when you are using the wrong classes. This does not occur in every instance, however, so we still recommend manually confirming that you are using the right object each time.
 
 ## China Map Styles
-Mapbox currently offers 3 government-certified map styles for China that match the look of our equivalent Mapbox Streets, Dark, and Light styles, but with the benefit of upto 10x faster map loading. You can either manually hardcode the style inside your app or use the provided constants found inside this Plugin. The table below lists the Java constant, XML string constant, and the actual URL which can be hardcoded in your app.
+Mapbox currently offers 3 government-certified map styles for China that match the look of our equivalent Mapbox Streets, Dark, and Light styles. The China styles provide up to 10x faster map loading. You can either manually hardcode the style URL inside your app or use the provided constants found inside this Plugin. The table below lists the Java constant, XML string constant, and the actual map style URL which can be hardcoded in your app.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: "Plugin" to "plugin"?

File: src/pages/plugins/overview/china.md [hunk 8]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
 | --- | --- |
 | `MapView` | `ChinaMapView` |
 | `MapboxMapOptions` | `MapboxMapChinaOptions` |
 | `MapboxDirections` | `MapboxDirectionsChina` |
 
-As mentioned above, Android Lint will attempt to warn you when you are using the wrong Map classes. This does not occur in every instance, however, so we still recommend manually confirming that you are using the right object each time.
+As mentioned above, Android Lint will attempt to warn you when you are using the wrong classes. This does not occur in every instance, however, so we still recommend manually confirming that you are using the right object each time.
 
 ## China Map Styles
-Mapbox currently offers 3 government-certified map styles for China that match the look of our equivalent Mapbox Streets, Dark, and Light styles, but with the benefit of upto 10x faster map loading. You can either manually hardcode the style inside your app or use the provided constants found inside this Plugin. The table below lists the Java constant, XML string constant, and the actual URL which can be hardcoded in your app.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: which --> that

File: src/pages/plugins/overview/china.md [hunk 9]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
 
 | Java Constant | XML String | URL |
 | --- | --- | --- |
 | `ChinaStyle.MAPBOX_STREETS_CHINESE` | `mapbox_style_mapbox_streets_chinese` | `mapbox://styles/mapbox/streets-zh-v1` |
 | `ChinaStyle.MAPBOX_LIGHT_CHINESE` | `mapbox_style_mapbox_dark_chinese` | `mapbox://styles/mapbox/light-zh-v1` |
 | `ChinaStyle.MAPBOX_DARK_CHINESE` | `mapbox_style_mapbox_light_chinese` | `mapbox://styles/mapbox/dark-zh-v1` |
 
-When using the constants found in this SDK you'll always be using the latest version of the map style when the plugin gets updated. Conversely, hardcoding allows you to have more control over the map style version and determining when your apps map style gets an update.
+When using the constants found in this SDK, you'll always be using the latest version of the map style when the plugin gets updated. Conversely, hardcoding allows you to have more control over the map style version and determining when your apps map style gets an update.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: determining --> determine

File: src/pages/plugins/overview/china.md [hunk 10]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
+We had to shift the base map for our default styles in order to comply with the Chinese government's mapping requirements. This means you are required to also shift any annotations that you place on top of your map so that they will match the shifted base map coordinate system. The plugin's shifting function will convert WGS-84 coordinates into GCJ-02 standards. We recommend reading [this wiki entry](https://en.wikipedia.org/wiki/Restrictions_on_geographic_data_in_China#The_China_GPS_shift_problem) and [this article](http://www.travelandleisure.com/articles/digital-maps-skewed-china) to better understand why shifting is required.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: >We had to shift

to 

>We shift

File: src/pages/plugins/overview/china.md [hunk 11]
```diff
@@ -48,44 +52,78 @@ Another option is to drop the architectures you do not need to support for your
...
+Do note that as mentioned above, the shift module is only available in the plugin's `China` flavor. This is the one exception to the rule mentioned above that you only need to switch dependency names if you'd like to support both global and China flavors of your app.
+
+In the case of a global version of your app, you'll need to remove the code that shifts your annotations. This will change in future releases of this plugin, so that shifting becomes much easier.
+
+
+## Shifting location
+
+[Read how to use the Mapbox `LocationEngine` in the Mapbox Core Libraries for Android](https://www.mapbox.com/android-docs/core/overview/#locationengine)
+
+When a new location update occurs, you'll need to manually feed in the `location` object into the plugin's `ShiftLocation` class' `shift()` method. Rather than dealing with raw coordinate values, this method and class handles `Location` objects. For now, the best way to do this is to create your own `LocationEngine` which extends another and listens for location updates. When the update occurs, send the `Location` object through the shift module and have the `locationEngine` provide the modified location.
```
Comment (colleenmcginnis) [inline src/pages/plugins/overview/china.md]: >feed ~~in~~ the `location` object into the plugin's...

don't need both in and into

File: src/pages/plugins/overview/china.md (hunk 1 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: Both fixed

File: src/pages/plugins/overview/china.md (hunk 2 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: fixed 🤦‍♂️

File: src/pages/plugins/overview/china.md (hunk 3 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: Fixed

File: src/pages/plugins/overview/china.md (hunk 4 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: Fixed

Comment (langsmith) [reply src/pages/plugins/overview/china.md]: fixed

File: src/pages/plugins/overview/china.md (hunk 7 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: Fixed

File: src/pages/plugins/overview/china.md (hunk 9 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: fixed

File: src/pages/plugins/overview/china.md (hunk 10 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: Fixed

File: src/pages/plugins/overview/china.md (hunk 11 above)
Comment (langsmith) [reply src/pages/plugins/overview/china.md]: Fixed
